base/lttagging.dtx

```diff
@@ -621,7 +644,215 @@
+% This socket is used in \cs{endsmallmatrix} to save the table. It takes no argument.
+% TODO: Check if this socket and the innertable socket can/should be merged into a more
+% generic version.
+%    \begin{macrocode}
+\NewSocket{tagsupport/math/luamml/mtable/smallmatrix/save}{0}
+%    \end{macrocode}
+% \end{socketdecl}
+% 
+% \begin{socketdecl}{tagsupport/math/luamml/mtable/innertable/finalize}
+% This socket is used e.g. in \cs{endsmallmatrix} and \cs{gathered} to finalize the table. 
+% It takes no argument.
+%    \begin{macrocode}
+\NewSocket{tagsupport/math/luamml/mtable/innertable/finalize}{0}
+%    \end{macrocode}
+% \end{socketdecl}
+% 
+% \begin{socketdecl}{tagsupport/math/luamml/mtable/tag/save}
+% This socket is used to save a tag for later use.
+% has been save before.
+%    \begin{macrocode}
+\NewSocket{tagsupport/math/luamml/mtable/tag/save}{0}
+%    \end{macrocode}
+% \end{socketdecl}
+
+% \begin{socketdecl}{tagsupport/math/luamml/mtable/tag/set}
+% This socket should be used when a tag is placed. It inserts a tag that
+% has been save before.
+%    \begin{macrocode}
+\NewSocket{tagsupport/math/luamml/mtable/tag/set}{0}
+%    \end{macrocode}
+% \end{socketdecl}
+% 
+% \paragraph{mbox socket}
+%
+% \begin{socketdecl}{tagsupport/math/luamml/hbox}
+%  This socket is used around \cs{hbox}
+%  inside an \cs{mbox}, \cs{makebox} and \cs{text} and annotates
+%  the content if the \cs{hbox} is used inside math.
+%  The real plug is set by luamml but a default plug is defined
+%  here so that the socket can also be used if luamml is not used.
+%    \begin{macrocode}
+\NewSocket{tagsupport/math/luamml/hbox}{2}
+\NewSocketPlug{tagsupport/math/luamml/hbox}{default}{#2}
+\AssignSocketPlug{tagsupport/math/luamml/hbox}{default}
+%    \end{macrocode}
+% \end{socketdecl}
+%
+%
+% \paragraph{math phantom sockets}
+% \changes{1.0m}{2025-01-27}{add sockets for math phantom commands}
+% \begin{socketdecl}{tagsupport/math/luamml/finph@nt}
+%  This socket handles the annotation of \cs{finph@nt}
+%    \begin{macrocode}
+\NewSocket{tagsupport/math/luamml/finph@nt}{2}
+\NewSocketPlug{tagsupport/math/luamml/finph@nt}{default}{#2}
+\AssignSocketPlug{tagsupport/math/luamml/finph@nt}{default}
+%    \end{macrocode}
+% \end{socketdecl}
+%
+% \begin{socketdecl}{tagsupport/math/luamml/finph@nt}
+%  This socket handles the annotation of \cs{finsm@sh}
+%    \begin{macrocode}
+\NewSocket{tagsupport/math/luamml/finsm@sh}{2}
+\NewSocketPlug{tagsupport/math/luamml/finsm@sh}{default}{#2}
+\AssignSocketPlug{tagsupport/math/luamml/finsm@sh}{default}
+%    \end{macrocode}
+% \end{socketdecl}
+% \paragraph{Artifact root sign}
+%
+% \begin{socketdecl}{tagsupport/math/luamml/artifact}
+% Unicode characters like a root sign should be marked as artifacts
+% to avoid duplication e.g. in derivation if mathml
+% structure elements are used that imply the meaning.
+%
+%    \begin{macrocode}
+\NewSocket{tagsupport/math/luamml/artifact}{0}
```

Comment:
wouldn't it be more logical if this socket takes the Unicode glyph (or whatever material) as an argument, i.e., have 2 args rather than 0?

Reply:
ok